app/lib/blocs/sign_up/signup_bloc.dart

```diff
@@ -0,0 +1,63 @@
+import 'package:effektio/blocs/sign_up/form_submition_status.dart';
+import 'package:effektio/blocs/sign_up/signup_event.dart';
+import 'package:effektio/blocs/sign_up/signup_state.dart';
+import 'package:effektio_flutter_sdk/effektio_flutter_sdk.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+
+class SignUpBloc extends Bloc<SignUpEvent, SignUpState> {
+  SignUpBloc() : super(SignUpState()) {
+    on<SignUpUsernameChanged>(_signUpUsername);
+    on<SignUpPasswordChanged>(_signUpPassword);
+    on<SignUpNameChanged>(_signUpName);
+    on<SignUpTokenChanged>(_signUpToken);
+    on<SignUpSubmitted>(_signUpSubmitted);
+  }
+
+  void _signUpUsername(SignUpUsernameChanged event, Emitter<SignUpState> emit) {
+    emit(state.copywith(username: event.username));
+  }
+
+  void _signUpPassword(SignUpPasswordChanged event, Emitter<SignUpState> emit) {
+    emit(state.copywith(password: event.password));
+  }
+
+  void _signUpToken(SignUpTokenChanged event, Emitter<SignUpState> emit) {
+    emit(state.copywith(token: event.token));
+  }
+
+  void _signUpName(
+    SignUpNameChanged event,
+    Emitter<SignUpState> emit,
+  ) {
+    emit(state.copywith(name: event.name));
+  }
+
+  Future<Client> signUp(
```

Comment:
Usually tokens get generated and not required by user to enter. Is this for testing purposes and will get removed eventually?